src/plugins/intel_cpu/src/nodes/memory.cpp

```diff
@@ -60,39 +57,159 @@ MemoryOutput::MemoryOutput(const std::shared_ptr<ov::Node>& op, const GraphConte
 }
 
+void MemoryOutput::initOptimalPrimitiveDescriptor() {
+    // Mimic the parent node memory desc to avoid extra reorder
+    auto parentEdge = getParentEdgeAt(0);
+    auto parent = parentEdge->getParent();
+    auto parentPd = parent->getSelectedPrimitiveDescriptor();
+    OPENVINO_ASSERT(parentPd,
+        parent->getTypeStr(), " ",
+        parent->getName(),
+        "failed getSelectedPrimitiveDescriptor() call, preferable primitive descriptor is not set");
+
+    const auto& parentConfig = parentPd->getConfig();
+    auto mem_desc = parentConfig.outConfs[parentEdge->getInputNum()].getMemDesc();
+
+    auto selected_pd = getSelectedPrimitiveDescriptor();
+    OPENVINO_ASSERT(selected_pd,
+        "MemoryOutput ",
+        getName(),
+        " failed getSelectedPrimitiveDescriptor() call, preferable primitive descriptor is not set");
+
+    auto config = selected_pd->getConfig();
+
+    const bool parentInplaceConflict = parent->inPlaceOutPort(parentEdge->getInputNum()) >= 0;
+
+    //disable inPlace to avoid inPlace conflict and handle memory copy internally (to get room for optimizations)
+    if (parentInplaceConflict) { config.inConfs.front().inPlace(-1); }
+    config.inConfs.front().setMemDesc(mem_desc);
+    //bypass any checks, we enforce the parent descriptor
+    selected_pd->setConfig(config);
+}
+
+void MemoryOutput::resolveInPlaceEdges(Edge::LOOK look) {
+    if (!(look & Edge::LOOK_DOWN)) {
+        Node::resolveInPlaceEdges(look);
+        return;
+    }
+
+    auto selected_pd = getSelectedPrimitiveDescriptor();
+    OPENVINO_ASSERT(selected_pd,
+        "MemoryOutput ",
+        getName(),
+        " failed getSelectedPrimitiveDescriptor() call, preferable primitive descriptor is not set");
+
+    auto parentEdge = getParentEdgeAt(0); // always only one parent edge
+
+    OPENVINO_ASSERT(one_of(parentEdge->getStatus(), Edge::Status::Uninitialized, Edge::Status::NotAllocated),
+        " Unexpected inplace resolve call to an allocated edge: ", parentEdge->name());
+
+    auto memDesc = selected_pd->getConfig().inConfs.front().getMemDesc();
+    memMngr = std::make_shared<ProxyMemoryMngr>();
+    auto edgeMem = std::make_shared<Memory>(getEngine(), memDesc, memMngr);
+    parentEdge->reuse(edgeMem);
+}
+
+void MemoryOutput::assignExtMemory(const MemoryPtr& mem, const MemoryDescPtr& memDesc) {
+    assignedMem = mem;
+    OPENVINO_ASSERT(assignedMem,
+        "MemoryOutput ",
+        getName(),
+        " assigned state has null memory ptr");
+
+    extMemDesc = memDesc;
+    OPENVINO_ASSERT(extMemDesc,
+        "MemoryOutput ",
+        getName(),
+        " assigned state has null base mem desc ptr");
+
+    if (!memMngr) { return; } //nothing to do, edge memory isn't under control
+    auto inpDesc = getBaseMemDescAtInputPort(0);
+
+    if (inpDesc->isCompatible(*extMemDesc)) {
+        memMngr->setMemMngrResize(assignedMem->getMemoryMngr());
```

Comment:
Why does it need setMemMngrResize instead of setMemMngr here? The resize stage of its parent node will resize it, won't it?

Reply:
The reason is the same as for the Output nodes memory. The resize step may be skipped if the `needShapeInferCall()` return false, so we may not rely on the parent node.